subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/annotations/RunFor.groovy

```diff
@@ -86,24 +89,44 @@ class RunForExtension implements IAnnotationDrivenExtension<RunFor> {
 
     static {
         Runtime.getRuntime().addShutdownHook(new Thread({
-            if (SCENARIO_DEFINITION_FILE != null) {
+            if (isGeneratingPerformanceScenarioDefinitions()) {
                 scenarioDefinition.writeTo(SCENARIO_DEFINITION_FILE)
             }
         } as Runnable))
     }
 
+    private static boolean isGeneratingPerformanceScenarioDefinitions() {
+        return SCENARIO_DEFINITION_FILE != null
+    }
+
     @Override
     void visitSpecAnnotation(RunFor runFor, SpecInfo spec) {
         assert runFor.value().every { it.iterationMatcher().isEmpty() }: "No iterationMatchers allowed in class-level @Scenario!"
         for (FeatureInfo feature : spec.getFeatures()) {
+            if (spec.getReflection().isAnnotationPresent(Ignore.class) || feature.getFeatureMethod().getReflection().isAnnotationPresent(Ignore.class)) {
```

Comment:
Shouldn't this go inside the next if? Since if `RunFor` is on the feature, then it would end up in `visitFeatureAnnotation`, right?